vendor/engines/secure_rooms/app/services/secure_rooms/access_rules/verdict.rb

```diff
@@ -0,0 +1,43 @@
+module SecureRooms
+
+  module AccessRules
+
+    class Verdict
+
+      attr_accessor :reason
+
+      def initialize(result_code, reason = nil)
+        @result_code = result_code
+        @reason = reason
+      end
+
+      def http_status
+        @http_status ||= status_for_code(@result_code)
+      end
+
+      def pass?
+        @result_code == :pass
+      end
+
+      def has_result_code?(code)
+        @result_code == code
+      end
+
+      private
+
+      def status_for_code(result_code)
```

Comment:
Part of me wonders if the HTTP translation should be part of the `Verdict` class or not. It sort of seems like a separate concern. I'm fine leaving it for now as we refactor further (I'd still prefer to see separate `GrantVerdict`, `DenyVerdict` and `PassVerdict` classes over this initialize with a symbol).